### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by Ghulam913-adil** [koalaman/shellcheck#2243]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1009, SC1073, SC1080, SC1010, SC1080, SC1072
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
if [#$==0]
then
        echo"Error: source And Destination Are Missing"
else if [#s==1]
then
        mv $1 ~/Desktop
        echo "$1 Is Moved To The ~/Desktop"
else if [#$==2]
then
        mv $1 $2
        echo "$1 Is Moved To $2"
else
        echo " Error: The Arguments Are More Than Expected"
fi

```

#### Here's what shellcheck currently says:
```
Line 1	SC1009: The mentioned syntax error was in this if expression.
Line 1	SC1073: Couldn't parse this test expression. Fix to allow more checks.
Line 1	SC1080: You need \ before line feeds to break lines in [ ].
Line 2	SC1010: Use semicolon or linefeed before 'then' (or quote to make it literal).
Line 2	SC1080: When breaking lines in [ ], you need \ before the linefeed.
Line 3	SC1072: Expected test to end here (don't wrap commands in []/[[]]). Fix any mentioned problems and try again.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

**Comment by charlescanato** [comment]
@Ghulam913-adil , your code doesn't even run here. Also, you didn't describe the output or any examples.

I would guess this is Bash, but then you might want to change your "else if"s to "elif"s, since Bash itself gets confused with this code and spits an "unexpected end of file" - makes sense, there aren't enough "fi"s.

**Comment by koalaman** [comment]
Assuming this was posted accidentally. If not, please replace TODOs and reopen
